lib/Runtime/Library/MathLibrary.cpp

```diff
@@ -714,35 +716,56 @@ namespace Js
             double result = JavascriptConversion::ToNumber(args[1], scriptContext);
             return JavascriptNumber::ToVarNoCheck(result, scriptContext);
         }
-        else if (args.Info.Count == 3)
+        else
         {
-            if (TaggedInt::Is(args[1]) && TaggedInt::Is(args[2]))
+            hasOnlyIntegerArgs = TaggedInt::OnlyContainsTaggedInt(args);
+            if (hasOnlyIntegerArgs && args.Info.Count == 3)
             {
                 return TaggedInt::ToVarUnchecked(max(TaggedInt::ToInt32(args[1]), TaggedInt::ToInt32(args[2])));
             }
         }
 
-        double current = JavascriptConversion::ToNumber(args[1], scriptContext);
-        if(JavascriptNumber::IsNan(current))
+        if (hasOnlyIntegerArgs)
         {
-            return scriptContext->GetLibrary()->GetNaN();
-        }
+            // **Assuming that TaggedInt cannot be NaN?
+            // Int32 vs Int64?
```

Comment:
And int64 is WASM only, which doesn't have these buijlt-ins.

---
In reply to: [131993866](https://github.com/Microsoft/ChakraCore/pull/3495#discussion_r131993866) [](ancestors = 131993866)